patterns/ink/InkRune.ts

```diff
@@ -86,18 +87,24 @@ export class InkRune<out C extends Chain, out U>
       .into(ExtrinsicRune, this.chain)
   }
 
+  // TODO: protect / consider whether this should be exposed
   filterContractEvents = <X>(...[events]: RunicArgs<X, [events: Event[]]>) => {
-    // TODO: return all relevant events, not just instantiated
     return Rune
       .tuple([Rune.resolve(events), this])
       .map(([events, publicKey]) =>
-        events.filter((e) =>
-          // TODO: clean up
-          isInstantiatedEvent(e) && equals((e.event as any).value.contract, publicKey)
+        events.filter((e): e is ContractsRuntimeEvent =>
+          isContractsRuntimeEvent(e)
+          && !!e.event.value.contract
+          && equals(e.event.value.contract, publicKey)
         )
       )
   }
 
+  // TODO: protect / consider whether this should be exposed
+  filterContractEmittedEvents = <X>(...[events]: RunicArgs<X, [events: Event[]]>) => {
+    return this.filterContractEvents(events).map((events) => events.filter(isContractEmitted))
+  }
+
   // TODO: improve
   decodeErrorEvent = <X>(...[failRuntimeEvent]: RunicArgs<X, [any]>) => {
```

Comment:
Should we rename this to `unhandleFailed` to keep it somewhat consistent with the extrinsic error unhandling? The difference ofc would be that one would pipe into this method (instead of it being accessible on an `ExtrinsicEventsRune`). Thoughts?